tabular/src/autogluon/tabular/models/fasttext/fasttext_model.py

```diff
@@ -0,0 +1,179 @@
+
+
+class FastTextModel(AbstractModel):
+    model_bin_file_name = "fasttext.ftz"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._model_bin_available = False
+
+    def _set_default_params(self):
+        default_params = get_param_baseline()
+        for param, val in default_params.items():
+            self._set_default_param_value(param, val)
+
+    def _get_default_auxiliary_params(self) -> dict:
+        default_auxiliary_params = super()._get_default_auxiliary_params()
+        extra_auxiliary_params = dict(
+            get_features_kwargs=dict(
+                required_special_types=[S_TEXT],
+            )
+        )
+        default_auxiliary_params.update(extra_auxiliary_params)
+        return default_auxiliary_params
+    
+    @classmethod
+    def _get_default_ag_args(cls) -> dict:
+        default_ag_args = super()._get_default_ag_args()
+        extra_ag_args = {'valid_stacker': False, 'problem_types': [BINARY, MULTICLASS]}
+        default_ag_args.update(extra_ag_args)
+        return default_ag_args
+
+
+    def _fit(self, X_train, y_train, **kwargs):
+        if self.problem_type not in (BINARY, MULTICLASS):
+            raise ValueError(
+                "FastText model only supports binary or multiclass classification"
+            )
+        
+        try_import_fasttext()
+        import fasttext
+
+        X_train = self.preprocess(X_train)
+        logger.debug("NLP features %s", self.features)
+
+        self._label_dtype = y_train.dtype
+        self._label_map = {label: f"__label__{i}" for i, label in enumerate(y_train.unique())}
+        self._label_inv_map = {v: k for k, v in self._label_map.items()}
+        idxs = np.random.permutation(list(range(len(X_train))))
+        mem_start = psutil.Process().memory_info().rss
+        with tempfile.NamedTemporaryFile(mode="w+t") as f:
+            logger.debug("generate training data")
+            for label, text in zip(y_train.iloc[idxs], (X_train[i] for i in idxs)):
+                f.write(f"{self._label_map[label]} {text}\n")
+            f.flush()
+            mem_start = psutil.Process().memory_info().rss
+            logger.debug("train FastText model")
+            self.model = fasttext.train_supervised(f.name, **self.params)
+            if kwargs.get('quantize_model', True):
+                self.model.quantize(input=f.name, retrain=True)
+            gc.collect()
+            mem_curr = psutil.Process().memory_info().rss 
+            self._model_size_estimate = mem_curr - mem_start
+            logger.debug("finish training FastText model")
+
+    # TODO: move logic to self._preprocess_nonadaptive()
+    def _preprocess(self, X: pd.DataFrame, **kwargs) -> list:
+        text_col = (
+            X[self.features]
+            .astype(str)
+            .fillna(" ")
+            .apply(lambda r: " __NEWCOL__ ".join(v for v in r.values), axis=1)
```

Comment:
does `__NEWCOL__` carry special meaning? Could this be shortened so it takes up less space (ex: ` _SEP_ `) if there are many columns? Could we just use a special single character?

Reply:
No. it has no special meaning. I just changed " __NEWCOL__ " to " " for now.

I initially had it as a simple way to deal with categorical variables so that tokens from categorical variables are in distinct context  than the same token from normal texts. 

Once we start to allow categorical values to be passed to text models, we can do some special encoding of categorical values. Another encoding is transform categorical features to <feature name>_<feature value>. We can see which one works better.